### ISSUE TITLE
Propagation on message queue systems

### ISSUE THREAD

**Issue by lukiano** [openzipkin/b3-propagation#33]
I know the x-b3-* headers are well defined for propagation through http, but what about propagation through message systems (for asynchronous producer / consumer spans)?
I think that in jms they use a b3 single header format, but I can't tell if this is standardized or not.

**Comment by codefromthecrypt** [comment]
you are right we don't limit messaging to just "b3" single because some systems don't have the same constraints as JMS. However, it is a very good choice for reasons of overhead and also compatibility. The best approach is lenient parsing.. meaning on the way in, parse both formats the same as HTTP mentions.

IOTW, while we mention this was originally designed for JMS, this doesn't mean only JMS implementation of messaging. Probably we should rename some sections.. do you want to give a go at doing that?

cc @openzipkin/core

**Comment by jcchavezs** [comment]
Just to mention, I am waiting for zipkin-go to merge single header to use
it in aws sqs propagation.

søn. 14. apr. 2019, 02:09 skrev Adrian Cole <notifications@github.com>:
